# Airport Cabs Data Analysis

#  

### Questions 

### In This Mini Numpy Project We Will Do Analysis On Airport's Cab Data 

#### Following Are The Questions That We Would Like To Get Answered

#### o  Mean Speed Of All Rides

#### o  Number Of Rides Taken In February

#### o  Number Of Rides Where Tip > 50$

#### o  Number Of Rides Where Drop Was JFK Airport

### Lets Start Our Analysis

#### First We Will The Numpy Library

In [1]:
import numpy as np

#### Now Lets Upload The CSV File To Out Jupyter Notebook

In [3]:
cab_data = np.genfromtxt('nyc_taxis.csv', delimiter=',', skip_header=True)

#### Lets Use Pandas Just For Showing The Data In A Pandas DataFrame

In [8]:
import pandas as pd

cab_data_pd = pd.read_csv('nyc_taxis.csv', dtype=str)
cab_data_pd.head(10)

pickup_year pickup_month pickup_day pickup_dayofweek pickup_time  \
0        2016            1          1                5           0   
1        2016            1          1                5           0   
2        2016            1          1                5           0   
3        2016            1          1                5           0   
4        2016            1          1                5           0   
5        2016            1          1                5           0   
6        2016            1          1                5           0   
7        2016            1          1                5           0   
8        2016            1          1                5           0   
9        2016            1          1                5           0   

  pickup_location_code dropoff_location_code trip_distance trip_length  \
0                    2                     4         21.00        2037   
1                    2                     1         16.29        1520   
2                    2                     6         12.70        1462   
3                    2                     6          8.70        1210   
4                    2                     6          5.56         759   
5                    4                     2         21.45        2004   
6                    2                     6          8.45         927   
7                    2                     6          7.30         731   
8                    2                     5         36.30        2562   
9                    6                     2         12.46        1351   

  fare_amount fees_amount tolls_amount tip_amount total_amount payment_type  
0       52.00        0.80         5.54      11.65        69.99            1  
1       45.00        1.30         0.00       8.00        54.30            1  
2       36.50        1.30         0.00       0.00        37.80            2  
3       26.00        1.30         0.00       5.46        32.76            1  
4       17.50        1.30         0.00       0.00        18.80            2  
5       52.00        0.80         0.00      52.80       105.60            1  
6       24.50        1.30         0.00       6.45        32.25            1  
7       21.50        1.30         0.00       0.00        22.80            2  
8      109.50        0.80        11.08      10.00       131.38            1  
9       36.00        1.30         0.00       0.00        37.30            2

#### As We Know To Find The Speed We Can Find It By Dividing The Trip Distance By Trip Length 
#### There Is A Little Problem As Trip Length Is Not In The Correct Unit Its In Seconds So We Will Convert It In HoursThen Calculate Speed That Should Be In Miles

In [9]:
speed = cab_data[:,7]/(cab_data[:,8]/3600)
mean_speed = speed.mean()
mean_speed

32.24258580925573

#### So Here Is Our First Answer The Mean Speed Is 32 mph

#### Now Lets Try To Answer Our 2nd Question Which Was To Find Number Of Rides Taken In The Month Of February

#### In Our Dataset Months Are Represented By Numbers So February Is Represented By 2

#### We Can FInd Our Answer By Using Boolean Algebra

In [18]:
rides_feb = cab_data[cab_data[:,1] == 2,1]
rides_feb

array([2., 2., 2., ..., 2., 2., 2.])

#### The rides_feb Variable Gives Us An Ndarray Of All The Rides Taken In The Month OF February 

#### But We Want The Count As Our Answer Not Array, We Can Get That By USing Shape Method

In [19]:
rides_feb.shape[0]

13333

#### Okay So Here Is Our 2nd Answer , The Number Of Rides Taken In The Month Of February Is 13333

#### Now Our Next Question Is To Calculate The Number Of Rides Where The Tip Paid Was Greater Then 50$

#### Here We Will Use Boolean Algebra Again

In [22]:
cab_data[cab_data[:,-3]>50].shape[0]

16

#### So Here Is Our Answer To The 3rd Question , Total 16 People Have Tipped More Than 50$

#### Now Our Last Question Is To Calculate The Number Of Rides Where The Drop Was At JFK Airport

#### In Our Dataset We Have Been Given Drop Of Location Code So It Makes All Sense To Use Boolean Algebra Again Here

In [24]:
cab_data[cab_data[:,6] ==2,6].shape[0]

11832

#### The Drop Of LOcation Code For JFK Airport Was 2 So Here Is Our Answer To The Last Question That Is The Total Number Of Rides Where Drop Of Location Was At JFK Airport Was 11832

#### So We Have Answered All Our Questions , Feel Free To Ask Your Own Questions And Answer Them Using Your Own Techniques

#### You Can Find The Dataset At https://drive.google.com/file/d/1Zu1xLzjcRKgWJ-B2k1j4BdMVFutkiqic/view 